### PR TITLE
Bug 1750654: The phase message of csc won't change when the env changed

### PR DESCRIPTION
**Pull request by awgreene**
Problem:
If a CatalogSourceConfig specifies a source that does not exist the
CatalogSourceConfig will get stuck in the configuring phase. This error
is encountered because the Reconcile method for the CatalogSourceConfig
is not returning an error, causing the handler not to reconcile the
CatalogSourceConfig.

Solution:
Update the CatalogSourceConfig Configuring Reconcile method to return an
error. Also adds e2e tests to prevent regressions in the future.

### CODE REVIEW

File: test/testsuites/packagestests.go [hunk 1]
```diff
@@ -197,6 +213,24 @@ func sourceMissingPackages(t *testing.T) {
 }
 
 func runSourceTest(namespace, source, packages, expectedPhase, expectedMessage string) error {
+	// Create the CatalogSourceConfig.
+	csc, err := createCscRuntimeObject(namespace, source, packages)
+	if err != nil {
+		return err
+	}
+
+	// Check that the CatalogSourceConfig with an non-existing targetNamespace eventually reaches the
```
Comment (gallettilance) [inline test/testsuites/packagestests.go]: typo: `a non-existent`

Comment (awgreene) [reply test/testsuites/packagestests.go]: Addressed in the latest commit.

File: test/testsuites/packagestests.go [hunk 2]
```diff
@@ -165,19 +165,35 @@ func unresolvedMissingSource(t *testing.T) {
 }
 
 // nonExistingSource ensures that if the provided source does not exist the CatalogSourceConfig
-// will be placed in the Configuring phase with the expected messages.
-func nonExistingSource(t *testing.T) {
+// will be placed in the Configuring phase with the expected messages. Then, the non-existant
```
[the remark this answers is not in the document]
Comment (awgreene) [reply test/testsuites/packagestests.go]: Addressed in the latest commit.

**Comment by awgreene** [comment]
[INFO] Acquiring a lease ...
failed to acquire a resource: Post http://boskos.ci/acquire?dest=leased&owner=ci-op-6yvtq2vc-bc8e3&request_id=2942913621423428907&state=free&type=aws-quota-slice: dial tcp 172.30.131.17:80: connect: no route to host

/test e2e-aws-console-olm

File: pkg/certificateauthority/handler.go [hunk 3]
```diff
@@ -42,7 +42,7 @@ func (h *configmapHandler) Handle(ctx context.Context, in *corev1.ConfigMap) err
 	caBundle := in.Data[caBundleKey]
 
 	// Retrieve the Certificate Authority bundle from Disk.
-	// If an error is returned and is not related to a nonexistant file, return an error.
+	// If an error is returned and is not related to a non-existent file, return an error.
```
Comment (kevinrizza) [inline pkg/certificateauthority/handler.go]: nit: we shouldn't include this change in this pr since it seems unrelated.

Comment (awgreene) [reply pkg/certificateauthority/handler.go]: Addressed.
